API/builder/targets/rpi2.py

```diff
@@ -40,6 +40,25 @@ def _build_freya(self):
         '''
         Cross-compile Valgrind and its Freya tool.
         '''
+        build_dir = utils.join(self.env['paths']['build'], 'valgrind_freya')
+        valgrind_files = [
+            'vg-in-place',
+            'coregrind/valgrind',
+            '.in_place/freya-arm-linux',
+            '.in_place/vgpreload_core-arm-linux.so',
+            '.in_place/vgpreload_freya-arm-linux.so'
+        ]
+
+        # Check if freya was already built, and skip the build if it was
+        built = True
+        for valgrind_file in valgrind_files:
+            if not utils.exists(utils.join(build_dir, valgrind_file)):
+                built = False
+                break
+
+        if built:
+            return
```

Comment:
I would create an `exist_files` (or something) named function in the `utils.py`. The parameter could be a base folder and a list (of files). In this case the code duplication could be eliminated.